Extension/src/LanguageServer/ui.ts

```diff
@@ -89,31 +86,94 @@ export class UI {
+            command: "C_Cpp.ShowReferencesProgress",
+            title: "",
+            tooltip: ""
+        };
+    }
+
+    private ensureIntellisentStatusItem(): void {
+        if (this.intelliSenseStatusBarItem) {
+            return;
+        }
+
+        this.intelliSenseStatusBarItem = vscode.languages.createLanguageStatusItem("c.cpp.intellisense.statusbar", this.documentSelector);
+        this.intelliSenseStatusBarItem.name = localize("c.cpp.intellisense.statusbar", "C/C++ IntelliSense Status");
+        this.intelliSenseStatusBarItem.detail = this.updatingIntelliSenseTooltip;
+    }
+
+    private ensureBrowseEnginerStatus(): void {
+        if (this.browseEngineStatusBarItem) {
+            return;
+        }
+
+        this.browseEngineStatusBarItem = vscode.languages.createLanguageStatusItem("c.cpp.tagparser.statusbar", this.documentSelector);
+        this.browseEngineStatusBarItem.name = localize("c.cpp.tagparser.statusbar", "C/C++ Tag Parser Status");
+        this.browseEngineStatusBarItem.detail = localize("discovering.files.tooltip", "Discovering files");
+    }
+
     private set ActiveConfig(label: string) {
         this.configStatusBarItem.text = label;
     }
 
     private set TagParseStatus(label: string) {
         this.workspaceParsingStatus = label;
-        this.browseEngineStatusBarItem.tooltip = (this.isParsingFiles ? `${this.parsingFilesTooltip} | ` : "") + label;
+
+        this.ensureBrowseEnginerStatus();
+        if (this.browseEngineStatusBarItem) {
+            this.browseEngineStatusBarItem.detail = (this.isParsingFiles ? `${this.parsingFilesTooltip} | ` : "") + label;
+        }
     }
 
     private setIsParsingWorkspace(val: boolean): void {
+        this.ensureBrowseEnginerStatus();
         this.isParsingWorkspace = val;
         const showIcon: boolean = val || this.isParsingFiles;
         const twoStatus: boolean = val && this.isParsingFiles;
         this.ShowDBIcon = showIcon;
-        this.browseEngineStatusBarItem.text = showIcon ? "$(database)" : "";
-        this.browseEngineStatusBarItem.tooltip = (this.isParsingFiles ? this.parsingFilesTooltip : "")
-            + (twoStatus ? " | " : "")
-            + (val ? this.workspaceParsingStatus : "");
+
+        if (this.browseEngineStatusBarItem) {
+            this.browseEngineStatusBarItem.text = showIcon ? "$(database)" : "";
+            this.browseEngineStatusBarItem.detail = (this.isParsingFiles ? this.parsingFilesTooltip : "")
+                + (twoStatus ? " | " : "")
+                + (val ? this.workspaceParsingStatus : "");
+        }
     }
 
     private setIsParsingWorkspacePausable(val: boolean): void {
-        if (val) {
-            this.browseEngineStatusBarItem.command = "C_Cpp.ShowParsingCommands";
-        } else {
-            this.browseEngineStatusBarItem.command = undefined;
+        this.ensureBrowseEnginerStatus();
+        if (this.browseEngineStatusBarItem) {
+            if (val) {
```

Comment:
nit: small refactoring opportunity -- rename all "val" variables to something descriptive.

Reply:
In short set/get functions, I think using something that means "value" is fine.  In C# you literally must use `value` in the `set` function since you don't get to name the variable.  The only alternative I see here would be `newValue` which isn't a significant improvement.